app/src/main/java/com/amaze/filemanager/filesystem/root/ListFilesCommand.kt

```diff
@@ -138,6 +160,18 @@ object ListFilesCommand : IRootCommand() {
         }
     }
 
+    /**
+     * Remaps given path to root-accessible paths. Returns null if no need to remap.
+     */
+    private fun remap(path: String): String? {
+        return if (!EXCLUDED_DIRS.contains(path)) {
+            null
+        } else {
+            val prefix = Environment.getExternalStorageDirectory().absolutePath
+            path.replace(prefix.substring(0, prefix.lastIndexOf('/')), "/data/media")
```

Comment:
I think `String.replace` is inefficient and will fail in the contrived example of `/storage/emulated/0/Android/data/com.example.badapp/storage/emulated/0/myfile.txt`.

Instead we can just replace the first occurrence, or better, assume that the first occurrence is at the start because it passed the EXCLUDED_DIRS conditional. The new expression would look something like `"/data/media" + path.substring(prefix.lastIndexOf('/'))`